src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl

```diff
@@ -0,0 +1,142 @@
+
+    return Fields.FieldVector(
+        nhm = Fields.FieldVector(ρ = ρ, ρuh = ρuh, ρw = ρw, ρθ = ρθ),
+    )
+end
+
+function Models.make_ode_function(model::Nonhydrostatic2DModel{FT}) where {FT}
+    @unpack p_0, R_d, cp_d, cv_d, g = model.parameters
+    γ = cp_d / cv_d
+
+    # TODO!: Replace with Thermodynamics.jl
+    pressure(ρθ) = ρθ >= 0 ? p_0 * (R_d * ρθ / p_0)^γ : NaN
+
+    # unity tensor for pressure term calculation 
+    # in horizontal spectral divergence
+    I = Ref(Geometry.Axis2Tensor(
+        (Geometry.Cartesian1Axis(), Geometry.Cartesian1Axis()),
+        @SMatrix [1.0]
+    ),)
+
+    # operators
+    # spectral horizontal operators
+    hdiv = Operators.Divergence()
+
+    # vertical FD operators with BC's
+    # interpolators
+    scalar_interp_c2f = Operators.InterpolateC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    vector_interp_c2f = Operators.InterpolateC2F(
+        bottom = Operators.SetValue(Geometry.Cartesian1Vector(0.0)),
+        top = Operators.SetValue(Geometry.Cartesian1Vector(0.0)),
+    )
+    tensor_interp_f2c = Operators.InterpolateF2C()
+
+    # gradients
+    scalar_grad_c2f = Operators.GradientC2F()
+    B = Operators.SetBoundaryOperator(
+        bottom = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+        top = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+    )
+
+    # divergences
+    vector_vdiv_f2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+        top = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+    )
+    tensor_vdiv_c2f = Operators.DivergenceC2F(
+        bottom = Operators.SetDivergence(Geometry.Cartesian3Vector(0.0)),
+        top = Operators.SetDivergence(Geometry.Cartesian3Vector(0.0)),
+    )
+    tensor_vdiv_f2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(
+            Geometry.Cartesian3Vector(0.0) ⊗ Geometry.Cartesian1Vector(0.0),
+        ),
+        top = Operators.SetValue(
+            Geometry.Cartesian3Vector(0.0) ⊗ Geometry.Cartesian1Vector(0.0),
+        ),
+    )
+
+    function rhs!(dY, Y, Ya, t)
+        dYm = dY.nhm
+        dρ = dYm.ρ
+        dρuh = dYm.ρuh
+        dρw = dYm.ρw
+        dρθ = dYm.ρθ
+        Ym = Y.nhm
+        ρ = Ym.ρ
+        ρuh = Ym.ρuh
+        ρw = Ym.ρw
+        ρθ = Ym.ρθ
+
+        # advection, pressure gradients, sources
+        # density
+        @. dρ = -hdiv(ρuh)
+        @. dρ -= vector_vdiv_f2c(ρw)
+
+        # potential temperature
+        @. dρθ = -hdiv(ρuh * ρθ / ρ)
+        @. dρθ -= vector_vdiv_f2c(ρw * scalar_interp_c2f(ρθ / ρ))
+
+        # horizontal momentum
+        @. dρuh = -hdiv(ρuh ⊗ ρuh / ρ + pressure(ρθ) * I)
+        @. dρuh -= tensor_vdiv_f2c(ρw ⊗ vector_interp_c2f(ρuh / ρ))
+
+        # vertical momentum
+        # TODO: How ugly is this?! (-_-)' . ￣□￣｜｜  (ಥ﹏ಥ)
+        uh_f = @. vector_interp_c2f(ρuh / ρ)
+        @. dρw = hdiv(uh_f ⊗ ρw)
```

Comment:
Comparing with [here](https://github.com/CliMA/ClimaCore.jl/blob/1a3c9463d367aa4dede9a11ffd81249ea8fa9017/examples/hybrid/bubble.jl#L198), should this be `@. dρw = -hdiv(uh_f ⊗ ρw)`?

Reply:
yes

Reply:
Good catch! Thanks!